src/app/_services/thumbnail.service.ts

```diff
@@ -0,0 +1,245 @@
+      }
+      // Use the compound media thumbnail url where sequenceNum equals 1
+      if (cleanedAsset && cleanedAsset.compound_media_json && cleanedAsset.compound_media_json.objects) {
+        let compoundAsset = cleanedAsset.compound_media_json.objects.filter((asset) => {
+          return asset['sequenceNum'] === 1
+        })
+        if (compoundAsset[0]) {
+          cleanedAsset.thumbnailUrls.push(this._auth.getThumbHostname(true) + compoundAsset[0].thumbnailSizeOnePath)
+        }
+      }
+      else { // make the thumbnail urls and add them to the array
+        for (let i = 1; i <= 5; i++) {
+          cleanedAsset.thumbnailUrls.push(this.makeThumbUrl(cleanedAsset, i))
+        }
+      }
+      // Set thumbnailImgUrl for multi-views
+      cleanedAsset.thumbnailImgUrl = cleanedAsset.thumbnailUrls[0]
+      // Attach media flags
+      cleanedAsset = this.attachMediaFlags(cleanedAsset)
+      // Attach usable thumbnail url
+      cleanedAsset.img = this.getThumbnailImg(cleanedAsset)
+
+      return <AssetThumbnail>cleanedAsset
+  }
+
+  /**
+   * Transforms object:
+   * Group/Item service result --> AIW AssetThumbnail
+   */
+  public itemAssetToThumbnail(item: RawItemAsset): AssetThumbnail 
+  {
+    // Map RawItemAsset fields to AssetThumbnail
+    let cleanedAsset: AssetThumbnail = {
+      id: item.objectId,
+      name: item.tombstone[0],
+      agent: item.tombstone[1],
+      date: item.tombstone[2],
+      objectTypeId: item.objectTypeId,
+      collectionTypeInfo: this.getCollectionType(item),
+      multiviewItemCount: item.compoundmediaCount ? item.compoundmediaCount : 0,
+      status: item.status,
+      img : '',
+      thumbnailImgUrl: item.thumbnailImgUrl,
+      zoom: item.zoom,
+      iap: item.iap,
+      partofcluster: item.clustered && item.clustered == 1,
+      clusterid: item.objectId,
+      frequentlygroupedwith: item.cfObjectId ? [item.cfObjectId] : []
+    }
+    // media takes priority over thumbnailImgUrl
+    if (item['media'] && item.media.thumbnailSizeOnePath) {
+      cleanedAsset.thumbnailImgUrl = item.media.thumbnailSizeOnePath
+    } else if (item['thumbnailImgUrl'] && item.compoundmediaCount > 0) {
+      cleanedAsset.thumbnailImgUrl = this._auth.getThumbHostname(true) + item.thumbnailImgUrl
+    }
+    // Attach media flags
+    cleanedAsset = this.attachMediaFlags(cleanedAsset)
+    // Attach usable thumbnail url
+    cleanedAsset.img = this.getThumbnailImg(cleanedAsset)
+    // Set as AssetThumbnail
+    return cleanedAsset
+  }
+
+  private getCollectionType(thumbnail: any): CollectionTypeInfo {
+    let collectionValue = thumbnail['collectionType'] ? [ thumbnail['collectionType'] ] : thumbnail['collectiontypes']
+    return CollectionTypeHandler.getCollectionType(collectionValue, thumbnail['contributinginstitutionid'])
+  }
+
+  /**
+   * Get image url to set on thumbnail
+   */
+  getThumbnailImg(thumbnail: AssetThumbnail): string {
```

Comment:
Do we want to stick this with the other public methods?

Reply:
This one should actually be private

Reply:
Just kidding! You're right